fairseq/utils.py

```diff
@@ -172,8 +172,8 @@ def make_positions(tensor, padding_idx, onnx_trace=False):
 
     Position numbers begin at padding_idx+1. Padding symbols are ignored.
     """
-    mask = tensor.ne(padding_idx).long()
-    return torch.cumsum(mask, dim=1) * mask + padding_idx
+    mask = tensor.ne(padding_idx).int()
+    return (torch.cumsum(mask, dim=1) * mask).long() + padding_idx
```

Comment:
This is causing unit tests to fail. This should work:
```
return (torch.cumsum(mask, dim=1, dtype=torch.int) * mask).long() + padding_idx
```

I'll update it and merge.

Reply:
Unfortunately I'm not able to make changes to the PR, so you'll need to update (or I can create a new PR). Either way, please confirm that the above works for XLA.

Reply:
Thanks Myle, I'm going to test this as soon as possible, however xla is broken atm due to https://github.com/pytorch/xla/issues/827, and it might take some time. I'll get back to you asap.